loopy/transform/iname.py

```diff
@@ -157,7 +157,62 @@ def map_variable(self, expr, expn_state):
             return super().map_variable(expr, expn_state)
 
 
-def _split_iname_backend(kernel, split_iname,
+def _split_iname_in_set(s, iname_to_split, inner_iname, outer_iname, fixed_length,
+        fixed_length_is_inner, split_iname_should_remain):
+    var_dict = s.get_var_dict()
+
+    if iname_to_split not in var_dict:
+        return s
+
+    orig_dim_type, _ = var_dict[iname_to_split]
+    assert orig_dim_type == dim_type.set
```

Comment:
The iname being split can be a param dim, as can be seen in the failing test. Allowing this flexibility (i.e. replacing all `dim_type.set` in the routine to `orig_dim_type`) passes the tests.

Reply:
fabd9c9f